### PR TITLE
Literal support as object

### PR DESCRIPTION
**Pull request by angelozerr**
Literal support as object

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

**Comment by angelozerr** [comment]
This PR uses java.lang.Integer instead of intfor literal support:

image

I need to clean my code and write some tests again.

File: qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/project/QuteProjectRegistry.java [hunk 1]
```diff
@@ -81,6 +84,32 @@ private static void registerPrimitiveType(String type) {
...
 				CompletableFuture.completedFuture(primitiveTypeArray));
+
+		String objectType = getPrmitiveObjectType(type);
+		if (objectType != null) {
+			autoboxing.put(type, type);
+			autoboxing.put(objectType, type);
+		}
+	}
+
+	private static String getPrmitiveObjectType(String type) {
```
Comment (fbricon) [inline qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/project/QuteProjectRegistry.java:95]: getPrimitiveObjectType

File: qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/project/QuteProjectRegistry.java [hunk 2]
```diff
@@ -81,6 +84,32 @@ private static void registerPrimitiveType(String type) {
...
+		case "char":
+			return "java.lang.Character";
+		case "double":
+			return "java.lang.Double";
+		case "float":
+			return "java.lang.Float";
+		case "int":
+			return "java.lang.Integer";
+		case "long":
+			return "java.lang.Long";
```
Comment (fbricon) [inline qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/project/QuteProjectRegistry.java:110]: maybe reference constants from LiteralSupport.java

**Review by fbricon — CHANGES_REQUESTED** [review]

File: qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/project/QuteProjectRegistry.java (hunk 1 above)
Comment (angelozerr) [reply qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/project/QuteProjectRegistry.java]: fixed

File: qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/project/QuteProjectRegistry.java (hunk 2 above)
Comment (angelozerr) [reply qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/project/QuteProjectRegistry.java]: Very good suggestion, I have moved the function in LiteralSupport class. fixed.

**Comment by fbricon** [comment]
Is it normal to still see `int` for {it} in this case?
```
{#each items.size}
  {it}
{/}
```

**Comment by angelozerr** [comment]
> Is it normal to still see int for {it} in this case?

Yes it's the type returned by JDT.

**Comment by fbricon** [comment]
Thanks @angelozerr

**Pull request merged**